### PR TITLE
Fixed #3592 - AOR Report- Date parameter doesn't work with any value of type 'Period'

### LINKED ISSUE #3592: AOR Report- Date parameter doesn't work with any value of type 'Period'
#### Issue

AOR Report- Date parameter doesn't work with any value of type 'Period'

#### Expected Behavior

Reports with a date parameter should display results 

#### Actual Behavior

Report returns empty and getting fatal errors in log 

    05/25/17 14:11:20 [2516][1][FATAL]  Query Failed:SELECT count(*) c FROM (
    SELECT [leads].do_not_call AS N'Do_Not_Call0', 
    [leads].first_name AS N'First_Name1', [leads].last_name AS N'Last_Name2', 
    [leads].id AS N'leads_id' 
    FROM [leads]  WHERE ([leads].date_entered BETWEEN "2017-05-24 00:00:00" AND "2017-05-25 14:11:20") 
    AND  leads.deleted = 0 ) as n::: 
    [Microsoft][SQL Server Native Client 10.0][SQL Server]Invalid column name '2017-05-24 00:00:00'.
    [Microsoft][SQL Server Native Client 10.0][SQL Server]Invalid column name '2017-05-25 14:11:20'.
    
    05/25/17 14:11:20 [2516][1][FATAL]   Query Failed:DECLARE @topCount INT SET @topCount = 20 
    SELECT TOP (@topCount)  [leads].do_not_call AS N'Do_Not_Call0', 
    [leads].first_name AS N'First_Name1', 
    [leads].last_name AS N'Last_Name2', [leads].id AS N'leads_id' FROM [leads]  
    WHERE (  [leads].date_entered BETWEEN "2017-05-24 00:00:00" AND "2017-05-25 14:11:20" ) 
    AND  leads.deleted = 0 ::: 
    [Microsoft][SQL Server Native Client 10.0][SQL Server]Invalid column name '2017-05-24 00:00:00'.
    [Microsoft][SQL Server Native Client 10.0][SQL Server]Invalid column name '2017-05-25 14:11:20'.

#### Possible Fix

Replace `"` (double quotes) with `'` (single quotes)

#### Steps to Reproduce

1. Create a report with a date parameter condition
2. Select `Period` as type and choose any value
3. Empty report is returned

#### Your Environment

* SuiteCRM Version used: 7.8.2
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): NA
* Environment name and version (e.g. MySQL, PHP 7): PHP 5.3, MSSQL Server 2008
* Operating System and version (e.g Ubuntu 16.04): Windows 7

### PR DESCRIPTION
**Pull request by ChangezKhan**
## Description

It solves issue #3592 

## Motivation and Context

Because of this change, one can create report consisting date parameter as condition with any value of type 'Period'

## How To Test This

1.    Create a report with a date parameter condition
2.    Select Period as type and choose any value
3.    Empty report is returned

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/AOR_Reports/AOR_Report.php [hunk 1]
```diff
@@ -1560,17 +1560,17 @@ function build_report_query_where($query = array())
                                 $params = base64_decode($condition->value);
                             }
                             $date = getPeriodEndDate($params)->format('Y-m-d H:i:s');
-                            $value = '"' . getPeriodDate($params)->format('Y-m-d H:i:s') . '"';
+                            $value = '\'' . getPeriodDate($params)->format('Y-m-d H:i:s') . '\'';
```
Comment (gunnicom) [inline modules/AOR_Reports/AOR_Report.php]: Why escaping and not just switch quotes like:
 `$value = "'" . getPeriodDate($params)->format('Y-m-d H:i:s') . "'";`

Comment (ChangezKhan) [inline modules/AOR_Reports/AOR_Report.php]: @gunnicom, I never thought about that. I will rework my code.

**Comment by Glen206** [comment]
I've tried the suggested edits but i'm still getting a fatal error and a blank report when using the type "period"

This is the error i'm currently getting after upgrading from 7.10.4

Fatal error: Uncaught Error: Call to a member function format() on null in //modules/AOR_Reports/AOR_Report.php:1562 Stack trace: #0 //modules/AOR_Reports/AOR_Report.php(1029): AOR_Report->build_report_query_where(Array) #1 //modules/AOR_Reports/AOR_Report.php(571): AOR_Report->build_report_query('', Array) #2 //modules/AOR_Reports/AOR_Report.php(556): AOR_Report->build_report_html(0, true, '', '8457d711-06c5-7...', Array) #3 
//modules/AOR_Reports/AOR_Report.php(338): AOR_Report->build_group_report(0, true, Array, '844fdc3d-56f0-b...') #4 
//modules/AOR_Reports/views/view.detail.php(78): AOR_Report->buildMultiGroupReport(0, true) #5 //include/MVC/View/SugarView.php(205): AOR_ReportsViewDetail->preDisplay() #6 //include/MVC/Controller/SugarController.php(432): SugarView->process() in //modules/AOR_Reports/AOR_Report.php on line 1562


Currently running:
SuiteCRM7.10.6
L
A
Mysql
Php 7.0

**Comment by jack7anderson7** [comment]
@ChangezKhan, is this error only occurring when the operator is "Equal To" or have you changed the operator when testing this? As when changing the operator to, for example, "Not Equal To" or "Greater Than" I don't appear to have an empty report returned, @gunnicom, was this the same for you or?

**Comment by gunnicom** [comment]
@jack7anderson7 Sorry, i can not remember. It is too long ago.

**Comment by gymad** [comment]
The problem because the `default` case is not handled in function `getPeriodEndDate()` in aor_utils:php

**Comment by Dillon-Brown** [comment]
Superseded by #6639

**Pull request closed without merging**